src/apps/ENGINE/file_service.h

```diff
@@ -101,14 +101,13 @@ class FILE_SERVICE : public VFILE_SERVICE
     std::time_t _ToTimeT(std::filesystem::file_time_type tp) override;
     std::filesystem::file_time_type _GetLastWriteTime(const char *filename) override;
     void _FlushFileBuffers(std::fstream &fileS) override;
-    uint32_t _GetCurrentDirectory(uint32_t nBufferLength, char *lpBuffer) override;
```

Comment:
I think this function should just return std::string. Passing a buffer to fill without passing the length of this buffer is a bad idea imo

Reply:
Done.